handbook/upgrade/upgrade.md

```diff
@@ -0,0 +1,280 @@
+
+1. Backup your SystemLink Server, MongoDB server or replica set, PostGreSQL server or replica set, and file store.
+
+    !!!note
+        If S3 is used as your file store the backup step is not needed since this managed internally by AWS.
+
+1. Attach a volume to store data captured by `nislmigrate`. This will be referenced as `D:\`.
+
+1. Install `nislmigrate`.
+
+1. Run the command `nislmigrate capture --tags --dir D:\migration`.
+
+    !!!note
+        While most of the data is external to SystemLink, tag current values are stored on the app server. Therefore this `nislmigrate` argument flag must be used to ensure that data is not lost during the migration.
+
+1. Detach `D:\`.
+
+    !!!note
+        At this point in the workflow your original SystemLink Server is in an operable state. Be aware if any new data is created or ingested by the SystemLink server at this time it will not be available to your new server.
+
+1. Provision a new Windows server for SystemLink.
+
+1. Provision a MongoDB server or replica set from the backup previously created.
+
+1. Provision a PostGres server or replica set from the backup previously created.
+
+    !!!note
+        The previous steps prescribe using a newly created instance of your MongoDB or PostGreSQL servers. This is because SystemLink may change the internal schema of these databases when services start post upgrade. If you do not start with a instance from a backup you will be unable to revert the MongoDB collections and PostGreSQL tables into the schemas needed for the older version of SystemLink.
+
+1. Install and configure the new version of SystemLink Server.
+
+1. Configure SystemLink to use the newly created MongoDB server or replica set, PostGreSQL server or replica set, and existing file store.
+
+1. Install `nislmigrate` on your new SystemLink Server.
+
+1. Attach the `D:\` volume used to capture data from your original SystemLink server.
+
+1. Run the command `nislmigrate restore --tags --dir D:\migration`.
+
+1. Verify your new SystemLink Server has all the expected migrated data.
+
+## Seamless Cut-over
+
+Since managed test systems are connected SystemLink it is desirable to approach upgrades or migrations such that test systems can connect to the new instance of SystemLink without manual intervention. To accomplish this the following conditions must be met.
+
+- Migration of systems data.
+
+    - This is either accomplished by using the `nislmigrate` argument flags `--all` or `--systems` or by connecting a MongoDB server or replica set that contains systems data.
+
+- The DNS name of the original SystemLink Server and the new SystemLink Server are the same.
+
+- If NI Web Server has been configured for HTTPS, the new SystemLink Server must use the same TLs certificate as the original SystemLink Server.
```

Comment:
Instructions for migrating this certificate?

Reply:
Do we need a certificate migrator?

Reply:
Let's leave this out of scope for now. Matt V mentioned that Josh has an example of how to do this in https://github.com/joshuaprewitt/systemlink-state-examples. I'm not sure which example does this. Regardless there is likely some cleanup work we would want to do on the example before providing this documentation. 

I created a new feature to cover the case of migrating a cert, or updating a cert and mass updating minions via a salt state. Lucky for us the need to support migration to K8s is going to encourage us to keep investing in the migration tool long term.